src/components/InvoiceItem.js

```diff
@@ -0,0 +1,29 @@
+import {useSelector} from "react-redux";
+import {selectInvoiceById} from "../store/invoicesSlice";
+
+const InvoiceItem = ({id}) => {
+
+    const allInvoices = useSelector(state => state.invoices.entities)
+    const invoice = useSelector((state) => selectInvoiceById(state, id))
+
+    const invoicesNumber = Object.keys(allInvoices).length;
+
+
+    if (invoicesNumber === 0) return <p>There is nothing here</p>
```

Comment:
Here my idea is to move this conditional about invoicesNumber to InvoiceList (if invoicesNumber === 0 return EmptyInvoiceList component). As this is connected with my task I can do that. Besides that I really like your code, it was very readable for me.